tests/proofs/dexter/dexter-spec.md

```diff
@@ -10,6 +10,38 @@ module DEXTER-SPEC
 
   claim <k> (now 0 => .) ~> #dexterCodeFA2 ... </k>
         <mynow> #Timestamp(0) </mynow>
+```
+
+## Set Manager
+
+The contract sets its manager to the provided manager address if the following conditions are satisfied:
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  exactly 0 tez was transferred to this contract when it was invoked
+3.  the txn sender is the `storage.manager`
+
+```
+  claim <k> #runProof(IsFA2, SetManagerParams(NewManager)) => . ... </k>
+        <manager> Sender => NewManager </manager>
+        <selfIsUpdatingTokenPool> true </selfIsUpdatingTokenPool>
+        <myamount> Amount </myamount>
+        <senderaddr> Sender </senderaddr>
+    ensures Mutez2Int(Amount) ==Int 0
+```
+
+If any of the conditions are not satisfied, the call fails.
+
+```
+  claim <k> #runProof(IsFA2, SetManagerParams(NewManager)) => #ExecutePostConditions ... </k>
+        <manager> CurrentManager </manager>
+        <selfIsUpdatingTokenPool> IsUpdating </selfIsUpdatingTokenPool>
+        <myamount> Amount </myamount>
+        <senderaddr> Sender </senderaddr>
+    ensures Mutez2Int(Amount) =/=Int 0
+     orBool Sender ==K CurrentManager
+     orBool IsUpdating)
```

Comment:
This approach won't work because `#ExecutePostConditions` is something that is setup as part of the semantics `#Init` routine which we don't use here (because we do a manual initialization).

In the original semantics, we use `#ExecutePostConditions` as a (incomplete yet cool) way of internalizing K's theorem proving (e.g. kprove) functionality into krun --- but we do not need that here, since we can directly use any kprove feature that we need (and if my experience tells me anything, it will work better that way).

The way to go in this case, I think, is to just let the `#Aborted` term consume the entire program, so that you end up with an `#Aborted(...)` term followed by other stuff.

Reply:
Another nit: I think the equality on line 41 should be an inequality.

Reply:
Hmm, do you think using `Aborted` will work? Looks like it. We need to make sure we have something which eats away everything on the <k> cell up to that point.

I guess we also need to make sure `Aborted` eats the `#storeDexterState` instruction and allows `#popState` instruction in order for the state to be otherwise unchanged. Perhaps we can use `<michelsonTop> _ => ?_ </michelsonTop>`